app/Main.hs

```diff
@@ -45,6 +46,16 @@ run options = do
 
     return ()
 
+retrieveResults :: Options -> [String] -> IO String
+retrieveResults options terms = do
+    cachedResult <- readCache
+    case cachedResult of
```

Comment:
I think this idea of "read, then calculate and write if missing" would fit better in `Unused.Cache`. In fact, such a function could be the only exported function of that module, leaving the implementation free to change if needed.

```hs
retrieveResults options terms = cached $ unlines -- ...

-- In Unused.Cache
cached :: IO String -> IO String
cached f = maybe (writeCache =<< f) return =<< readCache
```

Note that I changed `writeCache` to `String -> IO String` (e.g. to also return the value given), to make the above less syntactically awkward.

This `cached` function would also be more generically useful (IMO) if it took a first argument to represent the key, instead of calculating it internally -- but that could probably be saved for later if/when it becomes necessary.